partitura/musicanalysis/performance_features.py

```diff
@@ -0,0 +1,488 @@
+            n_notes = len(m_score)
+            if len(features) != n_notes:
+                msg = (
+                    "length of feature {} does not equal "
+                    "number of notes {}".format(len(features), n_notes)
+                )
+                raise InvalidPerformanceFeatureException(msg)
+
+            features_ = rfn.structured_to_unstructured(features)
+            if np.any(np.logical_or(np.isnan(features_), np.isinf(features_))):
+                problematic = np.unique(
+                    np.where(np.logical_or(np.isnan(features_), np.isinf(features_)))[1]
+                )
+                msg = "NaNs or Infs found in the following feature: {} ".format(
+                    ", ".join(np.array(features.dtype.names)[problematic])
+                )
+                raise InvalidPerformanceFeatureException(msg)
+
+            # prefix feature names by function name
+            feature_names = ["{}.{}".format(func.__name__, n) for n in features.dtype.names]
+            features = rfn.rename_fields(features, dict(zip(features.dtype.names, feature_names)))
+
+            acc.append(features)
+
+    if add_idx:
+        acc.append(np.array([(idx) for idx in snote_ids], dtype = [("id","U256")]))
+
+    performance_features = rfn.merge_arrays(acc, flatten=True, usemask=False)
+    full_performance_features = rfn.join_by("id", performance_features, m_score)
+    full_performance_features = full_performance_features.data
+    pitch_sort_idx = np.argsort(full_performance_features["pitch"])
+    full_performance_features = full_performance_features[pitch_sort_idx]
+    onset_sort_idx = np.argsort(full_performance_features["onset"], kind="mergesort")
+    full_performance_features = full_performance_features[onset_sort_idx]
+    return  full_performance_features
+
+
+def compute_matched_score(score: ScoreLike, 
+                            performance: PerformanceLike,
+                            alignment: list,):
+    """
+    Compute the matched score and add the score features
+
+    Parameters
+    ----------
+    score : partitura.score.ScoreLike
+        Score information, can be a part, score
+    performance : partitura.performance.PerformanceLike
+        Performance information, can be a ppart, performance
+    alignment : list
+        The score--performance alignment, a list of dictionaries
+
+    Returns
+    -------
+    m_score : np strutured array
+    unique_onset_idxs : list
+    """
+
+    m_score, snote_ids = to_matched_score(score, performance, alignment, include_score_markings=True)
+    (time_params, unique_onset_idxs) = encode_tempo(
+        score_onsets=m_score["onset"],
+        performed_onsets=m_score["p_onset"],
+        score_durations=m_score["duration"],
+        performed_durations=m_score["p_duration"],
+        return_u_onset_idx=True,
+        tempo_smooth="average"
+    )
+    m_score = rfn.append_fields(m_score, "beat_period", time_params['beat_period'], "f4", usemask=False)
```

Comment:
It might be better to use a single call to  `rfn.append_fields`. Maybe something like 

```python
m_score = rfn.append_fields(m_score, ["beat_period", "id"], [time_params['beat_period'], snote_ids], ["f4", "U256"], usemask=False,)
```

Reply:
addressed in commit [e6f7124](https://github.com/CPJKU/partitura/pull/260/commits/e6f7124610b7886ae13378b60d2d6ccd32401fa1)